domain-ee/ee-ep-merge-app/src/python_src/service/ep_merge_machine.py

```diff
@@ -274,17 +279,49 @@ def on_get_ep400_contentions(self, event, pending_contentions_response=None):
 
         self.send(event=event, pending_contentions_response=pending_contentions_response, ep400_contentions_response=response)
 
+    def verify_ep400_contention_special_issue_codes(self, ep400_contentions: list[ContentionSummary], pending_contentions: list[ContentionSummary]) -> None:
+        request = get_special_issue_types.Request()
+        response = self.make_request(
+            request=request, hoppy_client=HOPPY.get_client(ClientName.GET_SPECIAL_ISSUE_TYPES), response_type=get_special_issue_types.Response
+        )
+
+        if response and response.status_code == 200:
+            deactive_special_issue_codes = [
+                special_issue.code
+                for special_issue in response.code_name_pairs or []
+                if special_issue.deactive_date and special_issue.deactive_date < datetime.now(special_issue.deactive_date.tzinfo)
+            ]
+            if deactive_special_issue_codes:
+                mergeable_contentions = ContentionsUtil.new_contentions(pending_contentions, ep400_contentions)
+                deactive_codes_on_mergeable_contentions = [
+                    code
+                    for contention in (mergeable_contentions or [])
+                    for code in (contention.special_issue_codes or [])
+                    if code in deactive_special_issue_codes
+                ]
+
+                if deactive_codes_on_mergeable_contentions:
+                    self.add_job_error(
+                        f'EP400 claim #{self.job.ep400_claim_id} has contentions to merge that have the following deactive special issue codes: '
+                        f'{", ".join(deactive_codes_on_mergeable_contentions)}'
+                    )
+            else:
+                self.add_job_error(
+                    f'Could not retrieve special issue codes to verify EP400 claim #{self.job.ep400_claim_id} does not contain contentions with '
+                    f'deactive special issue codes'
+                )
```

Comment:
Do you not need to do a `self.send()` in this `@state.enter` method?

Reply:
The method `verify_ep400_contention_special_issue_codes` is called inside of the `@running_get_ep400_contentions.enter` method which also calls `self.send()` at the bottom 👍